Source/WebCore/platform/gstreamer/GStreamerQuirks.h

```diff
@@ -87,6 +87,11 @@ class GStreamerQuirk : public GStreamerQuirkBase {
     virtual int correctBufferingPercentage(MediaPlayerPrivateGStreamer*, int originalBufferingPercentage, GstBufferingMode) const { return originalBufferingPercentage; }
     virtual void resetBufferingPercentage(MediaPlayerPrivateGStreamer*, int) const { };
     virtual void setupBufferingPercentageCorrection(MediaPlayerPrivateGStreamer*, GstState, GstState, GRefPtr<GstElement>&&) const { }
+    virtual bool needsCustomInstantRateChange() const { return false; }
+    virtual bool applyCustomInstantRateChange(bool isPipelinePlaying, bool isPipelineWaitingPreroll, float playbackRate,
+        bool mute, GstElement* pipeline, bool& didInstantRateChange) const;
+    virtual void analyzeWebKitMediaSrcCustomEvent(GRefPtr<GstEvent>, bool& forwardToAllPads) const;
+    virtual void processWebKitMediaSrcCustomEvent(GRefPtr<GstEvent>, bool handledByAnyStream, bool handledByAllTheStreams, double& rate) const;
```

Comment:
We don't use output parameters anymore. We should use return `std::pair<bool, bool>` for apply and then bool and double for the analyze and process.

Reply:
Would you accept something like this as a way to document the members in the pair?
```
std::pair<bool /* processed */, bool /* didInstantRateChange */> applyCustomInstantRateChange(
    bool isPipelinePlaying, bool isPipelineWaitingPreroll, float playbackRate,
    bool mute, GstElement* pipeline) const;
```

Reply:
Sure, or even a struct that contains the two bools with their names. That's also self documenting.